ownCloud/Viewer/ImageDisplayViewController.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  ImageDisplayViewController.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 17/10/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+
+class ImageDisplayViewController : DisplayViewController {
+
+	var scrollView: UIScrollView?
+	var imageView: UIImageView?
+
+	override func renderSpecificView() {
+		scrollView = UIScrollView()
+
+		scrollView?.translatesAutoresizingMaskIntoConstraints = false
+		scrollView?.maximumZoomScale = 6.0
+		scrollView?.bounces = false
+		scrollView?.bouncesZoom = false
+		scrollView?.delegate = self
+		self.view.addSubview(scrollView!)
+
+		NSLayoutConstraint.activate([
+			scrollView!.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor),
+			scrollView!.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor),
+			scrollView!.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+			scrollView!.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor)
+
+		])
+
+		imageView = UIImageView()
+		do {
+			let data = try Data(contentsOf: source)
+			let image = UIImage(data: data)
+			imageView?.image = image
+			imageView?.contentMode = .scaleAspectFit
+			imageView?.translatesAutoresizingMaskIntoConstraints = false
+			imageView?.backgroundColor = .black
+			scrollView!.addSubview(imageView!)
+			NSLayoutConstraint.activate([
+				imageView!.centerXAnchor.constraint(equalTo: scrollView!.centerXAnchor),
+				imageView!.centerYAnchor.constraint(equalTo: scrollView!.centerYAnchor),
+				imageView!.leftAnchor.constraint(equalTo: scrollView!.leftAnchor),
+				imageView!.rightAnchor.constraint(equalTo: scrollView!.rightAnchor),
+				imageView!.bottomAnchor.constraint(equalTo: scrollView!.bottomAnchor),
+				imageView!.topAnchor.constraint(equalTo: scrollView!.topAnchor)
```

Comment:
Using constraints to center the imageView inside the scrollview works initially, but the zoom will also scale up the blank space around the image later. Scrolling around while zoomed into the image, then, can give you this result:
<img width="545" alt="bildschirmfoto 2018-10-17 um 23 20 18" src="https://user-images.githubusercontent.com/639669/47116934-51c10680-d263-11e8-9794-9824952d60d1.png">

You can use `UIScrollView.contentInsets` and adapt these in `UIScrollViewDelegate`'s `scrollViewDidZoom(_ scrollView: UIScrollView)` as zoom levels change to increase or decrease insets value. [This Stackoverflow question](https://stackoverflow.com/questions/14069571/keep-zoomable-image-in-center-of-uiscrollview) gives a few examples on how that could be done.